engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -212,14 +202,14 @@ object PartialWorkflowSources {
     partialSources match {
       case Valid(partialSource) =>
         (validateInputs(partialSource),
-          validateOptions(partialSource.workflowOptions), validateWorkflowType(partialSource),
-          validateWorkflowTypeVersion(partialSource), validateLabels(partialSource.customLabels.getOrElse("{}"))) mapN {
-          case (wfInputs, wfOptions, workflowType, workflowTypeVersion, workflowLabels) =>
+          validateOptions(partialSource.workflowOptions),
+          validateLabels(partialSource.customLabels.getOrElse("{}"))) mapN {
```

Comment:
this `getOrElse` looks like a code smell.

can this `getOrElse` be put into a constructor for partial source, then the type of customLabels just be `WorkflowJson`?  Or maybe this validate code could be put into that class?

Reply:
Having re-looked at it only briefly, the `PartialWorkflowSources` class seems to accumulate fields (eg from a REST request) which may or may not be incomplete/inconsistent, and then this `validate` method turns it into a guaranteed complete version that the rest of the process uses.

Personally, I'm sort of inclined to think that this "validate that this partial set is good enough and return a validated set" function is a good enough place to inject a default in case the REST request didn't include a `customLabels` field?

Reply:
It might be (it definitely is) super-paranoid - but if the partial sources had `{}` as the default on that field, we couldn't tell whether we incorrectly got two copies of `customLabels` in the request - though I'm not sure that's actually checked anyway...